app/models/attachment.rb

```diff
@@ -2,7 +2,7 @@
 class Attachment < ApplicationRecord
   belongs_to :parent, polymorphic: true, inverse_of: :attachments, touch: true
 
-  before_update { raise ActiveRecord::ReadOnlyRecord } # Prevent updates
+  before_update { raise ActiveRecord::ReadOnlyRecord if changed? } # Prevent updates
```

Comment:
I'm not sure when `before_update` would fire when there aren't changes

Reply:
Actually I think the issue was the framework defaults for Rails 6.1, not Ruby 3.1